R-package/R/lgb.Dataset.R

```diff
@@ -774,7 +776,8 @@ lgb.Dataset <- function(data,
 #' @title Construct validation data
 #' @description Construct validation data according to training data
 #' @param dataset \code{lgb.Dataset} object, training data
-#' @param data a \code{matrix} object, a \code{dgCMatrix} object or a character representing a filename
+#' @param data a \code{matrix} object, a \code{dgCMatrix} object or
+#'             a character representing a path to a text file (CSV, TSV, or LibSVM)
```

Comment:
In Python, it can also be a binary file. I guess it's true and for the R-package.
```py
import lightgbm as lgb
from sklearn.datasets import load_boston

X, y = load_boston(return_X_y=True)
lgb_data = lgb.Dataset(X, y)
lgb_data.save_binary('dataset.bin')
lgb_valid = lgb_data.create_valid('dataset.bin')
lgb.train({'objective': 'regression'}, lgb_valid, num_boost_round=5)
lgb_valid.get_field('label')
```

Reply:
updated in https://github.com/microsoft/LightGBM/pull/4545/commits/33448b6d2a396447e6515b8357d69fa1cfc7c351, thanks for the sample code!